src/System.Text.Json/src/System/Text/Json/Node/JsonNode.cs

```diff
@@ -31,7 +29,10 @@ public abstract class JsonNode
         /// </summary>
         /// <param name="jsonElement"><see cref="JsonElement"/> to get the <see cref="JsonNode"/> from.</param>
         /// <returns><see cref="JsonNode"/> represented by <paramref name="jsonElement"/>.</returns>
-        public static JsonNode GetNode(JsonElement jsonElement) => (JsonNode)jsonElement._parent;
+        /// <exception cref="ArgumentException">
+        ///   Provided <see cref="JsonElement"/> was not build from <see cref="JsonNode"/>.
+        /// </exception>
+        public static JsonNode GetNode(JsonElement jsonElement) => !jsonElement.IsImmutable ? (JsonNode)jsonElement._parent : throw new ArgumentException(SR.NotNodeJsonElementParent);
```

Comment:
Did we ever discussed why is this one called GetNode and not GetJsonNode? Giving the full name of the return type would be better in my opinion.

Reply:
We didn't discuss it; yes, probably `GetJsonNode` would be better.